src/ts/circular_buffer.ts

```diff
@@ -63,7 +63,14 @@ class circular_buffer<T> {
   }
 
 
-
+  fill(x:any): T[] {
+    for (var i = 0; i < this._capacity; i++) {
+      this._values[i] = x;
+      this._length = i;
```

Comment:
length can be after the loop once, instead of every step in the loop.  javascript is single threaded; there's no chance of something else reading the information while it's working

Reply:
roger that.  That part is my fault.  I remember you telling me and I didn't do it right away, ultimately forgetting that -that was a thing to do.